app/src/main/java/com/duckduckgo/app/history/HistoryEntry.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.history
+
+import android.net.Uri
+import androidx.core.net.toUri
+import com.duckduckgo.app.history.HistoryEntry.VisitedPage
+import com.duckduckgo.app.history.HistoryEntry.VisitedSERP
+import com.duckduckgo.app.history.store.HistoryEntryWithVisits
+import java.util.Date
+
+sealed class HistoryEntry(
```

Comment:
I have a similar question again: Visiting a site or performing a search, are both visited pages. So having 2 models is telling me we will return one or the other, and that we see them as different.

My expectations is that I can query visited sites and I receive all visited sites (including SERP urls); and I can query searches and I only receive searches the use has made.

Does it make sense?